master/buildbot/newsfragments/master-bug-2542.bugfix

```diff
@@ -0,0 +1 @@
+[GitPoller] Ignore missing branches on remote. Only attempt to fetch branches known to exist on remote. Non-existing branches are quietly ignored
```

Comment:
We don't use prefixes like `[GitPoller]` in release notes. I suggest shortening to something like:

`Gitpoller now fetches only branches that are known to exist on remote. Non-existing branches are quietly ignored.`